### PR TITLE
Expect:Continue implementation for Client

### PR DESCRIPTION
**Pull request by shafreenAnfar**

### CODE REVIEW

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contractimpl/sender/states/SendingHeaders.java [hunk 1]
```diff
@@ -125,8 +147,15 @@ private void setHttpVersionProperty(HttpCarbonMessage httpOutboundRequest, Strin
     }
 
     private void writeRequestBody(HttpCarbonMessage httpOutboundRequest, HttpContent httpContent) {
-        senderReqRespStateManager.state =
-                new SendingEntityBody(senderReqRespStateManager, httpInboundResponseFuture);
+        String expectHeader = httpOutboundRequest.getHeader(HttpHeaderNames.EXPECT.toString());
+        if (expectHeader != null && !expectHeader.isEmpty() && expectHeader.equalsIgnoreCase(HEADER_VAL_100_CONTINUE)) {
```
Comment (chamil321) [inline components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contractimpl/sender/states/SendingHeaders.java]: Non empty check is unnecessary as we equals the header value with `HEADER_VAL_100_CONTINUE`

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contractimpl/sender/states/SendingHeaders.java [hunk 2]
```diff
@@ -125,8 +147,15 @@ private void setHttpVersionProperty(HttpCarbonMessage httpOutboundRequest, Strin
...
+            senderReqRespStateManager.state =
+                    new Sending100Continue(senderReqRespStateManager, httpInboundResponseFuture);
+            senderReqRespStateManager.nettyTargetChannel.flush();
```
Comment (chamil321) [inline components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contractimpl/sender/states/SendingHeaders.java]: Should not we return from here?

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contractimpl/sender/states/Sending100Continue.java [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+ * This class implements SenderStates. The responsibility of this class is to handle requests with
+ * Expect Continue header. As in the spec, it should only send headers of the request and wait for server response
+ * before sending the request body.
+ */
+public class Sending100Continue implements SenderState {
+
+    private static final Logger LOG = LoggerFactory.getLogger(Sending100Continue.class);
+
+    private final SenderReqRespStateManager senderReqRespStateManager;
+
```
Comment (chamil321) [inline components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contractimpl/sender/states/Sending100Continue.java]: unnecessary newline

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contractimpl/sender/states/SendingHeaders.java [hunk 4]
```diff
@@ -168,6 +192,8 @@ public void handleAbruptChannelClosure(HttpResponseFuture httpResponseFuture) {
     @Override
     public void handleIdleTimeoutConnectionClosure(HttpResponseFuture httpResponseFuture, String channelID) {
         // HttpResponseFuture will be notified asynchronously via writeOutboundRequestHeaders method.
```
Comment (chamil321) [inline components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contractimpl/sender/states/SendingHeaders.java]: Shall we remove this comment as well?

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contractimpl/sender/states/SendingHeaders.java (hunk 1 above)
Comment (shafreenAnfar) [reply components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contractimpl/sender/states/SendingHeaders.java]: Done

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contractimpl/sender/states/Sending100Continue.java (hunk 3 above)
Comment (shafreenAnfar) [reply components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contractimpl/sender/states/Sending100Continue.java]: This was intentional :)

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contractimpl/sender/states/SendingHeaders.java (hunk 4 above)
Comment (shafreenAnfar) [reply components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contractimpl/sender/states/SendingHeaders.java]: Can't remove it yet till we properly test.

**Pull request merged**